tests/usm/usm_atomic_access.h

```diff
@@ -0,0 +1,173 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides common code for USM atomic access tests
+//
+*******************************************************************************/
+
+#ifndef __SYCL_CTS_TEST_USM_ATOMIC_ACCESS_H
+#define __SYCL_CTS_TEST_USM_ATOMIC_ACCESS_H
+
+#include "../../util/usm_helper.h"
+#include "../common/type_coverage.h"
+#include <chrono>
+#include <cstring>
+#include <limits.h>
+#include <thread>
+#include <type_traits>
+
+namespace usm_atomic_access {
+
+static auto get_scalar_types() {
+  static const auto scalar_types =
+      named_type_pack<short, unsigned short, int, unsigned int, long,
+                      unsigned long, float, double, long long,
+                      unsigned long long>({"short", "unsigned short", "int",
+                                           "unsigned int", "long",
+                                           "unsigned long", "float", "double",
+                                           "long long", "unsigned long long"});
+  return scalar_types;
+}
+
+/** @brief Return atomic aspect depending on the type of allocated memory
+ *  @retval AllocMemT USM allocation type
+ */
+template <sycl::usm::alloc AllocMemT>
+auto get_atomic_memory() {
+  if constexpr (AllocMemT == sycl::usm::alloc::host) {
+    return sycl::aspect::usm_atomic_host_allocations;
+  } else if constexpr (AllocMemT == sycl::usm::alloc::shared) {
+    return sycl::aspect::usm_atomic_shared_allocations;
+  } else {
+    static_assert(AllocMemT != AllocMemT,
+                  "This USM allocation type does not supported");
+  }
+}
+
+using with_atomic64 = std::true_type;
+using without_atomic64 = std::false_type;
+/** @brief Dummy functor that use in type coverage
+ *  @tparam CounterT Variable type from type coverage
+ *  @tparam UseAtomic64Flag std::integral_constant type
+ */
+template <typename CounterT, typename UseAtomic64Flag>
```

Comment:
```suggestion
template <typename CounterT, bool UseAtomic64Flag>
```
looks clearer, with
```
auto constexpr witth_atomic64 = true;
```
and so on?

Reply:
`bool` type cannot be provided to the `for_all_types` function. by this function I call `run_all_tests` in 
[usm_atomic_access_core.cpp](https://github.com/KhronosGroup/SYCL-CTS/pull/158/files/7d316d63dd0cdd9b111c7f41a0386db167502b02#diff-29697b92aebee3800e595593638621294b090a5f860c125ba20694b76373188a) and [usm_atomic_access_atomic64.cpp](https://github.com/KhronosGroup/SYCL-CTS/pull/158/files/7d316d63dd0cdd9b111c7f41a0386db167502b02#diff-dc3e9b7188bf1db340a2b583845b1da7e8bdb2266fa0eb6c67e6258a139a75be)

Reply:
I guess you would have to propagate the API change in the usage and the callers too.
Or is there another issue I do not see?

Reply:
`for_all_types` used across all project and this changes requir do some changes in many files and then verification that changed tests works correctly
I suggest did not do this changes and just provide this argument in common parameters, as implemented in this commit: a73c49c988826254f402716a6e909aef9c041ad5